src/qiskit_sphinx_theme/__init__.py

```diff
@@ -78,6 +80,28 @@ def activate_themes(app: sphinx.application.Sphinx, config: sphinx.config.Config
         app.setup_extension("sphinxcontrib.jquery")
 
 
+def override_furo_toc(
+    app: sphinx.application.Sphinx,
+    pagename: str,
+    templatename: str,
+    context: dict[str, Any],
+    doctree: sphinx.addnodes.document,
+) -> None:
+    """Furo always fully expands its table of contents, which takes way too long to build docs
+    for most Qiskit projects."""
+    if "toctree" in context:
```

Comment:
do we need to copy this whole chunk of code from fruo? is it possible for us to just set the maxdepth param of the toctree and leave the others to be set by furo? or do we have to overwrite the entire toctree object?

Reply:
Yeah, it's necessary to copy it all. I posted upstream if he'd be willing to accept a PR for adding `navigation_depth` since it would only be about 5 lines.